jib-core/src/main/java/com/google/cloud/tools/jib/regex/RegexBuilder.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.regex;
+
+import java.util.regex.Matcher;
+
+/**
+ * Static methods for constructing a regular expression in a more readable functional form. Extend
+ * this class to use the static methods.
+ */
+public abstract class RegexBuilder {
```

Comment:
There's some really interesting ideas in here, but this abstract class with static methods seems strange. Why wouldn't you just use the utility class pattern instead?

Reply:
The main reason was to be able to encapsulate regex constants in a child class and also be able to have that class construct the regex using the static methods without statically importing the methods for visibility to others.